flink/src/test/java/org/apache/iceberg/flink/data/TestFlinkAvroReaderWriter.java

```diff
@@ -22,52 +22,76 @@
 import java.io.File;
 import java.io.IOException;
 import java.util.Iterator;
-import org.apache.flink.types.Row;
+import java.util.List;
+import org.apache.flink.table.data.RowData;
 import org.apache.iceberg.Files;
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.data.DataTest;
+import org.apache.iceberg.data.RandomGenericData;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.data.avro.DataWriter;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.junit.Assert;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
 
-import static org.apache.iceberg.flink.data.RandomData.COMPLEX_SCHEMA;
+public class TestFlinkAvroReaderWriter extends DataTest {
 
-public class TestFlinkAvroReaderWriter {
-  private static final int NUM_RECORDS = 20_000;
+  private static final int NUM_RECORDS = 100;
 
-  @Rule
-  public TemporaryFolder temp = new TemporaryFolder();
+  @Override
+  protected void writeAndValidate(Schema schema) throws IOException {
+    List<RowData> inputs = generateDataFromAvroFile(schema);
```

Comment:
I see you will generate the `List<Record>` firstly, then write to the file appender, and finally read them into `List<RowData>`,   could we just use the `RandomData#generateRowData`  to produce those `RowData` ?

Reply:
- First, `RandomData` now is incorrect, like array, like timestamp with zone, and etc.. 
- Second, using Iceberg avro writer can test format compatible better.